### PR TITLE
Fix/1524  make config paths mandatory

### PR DESCRIPTION
**Pull request by inkhey**
need #1661 , this fix also #1083 
related to #1524 
more info about content: https://github.com/tracim/tracim/issues/1524#issuecomment-490919128
Todo: 
- [X]  Fixed ! ~~coverall seems completly broken now (It was already partially broken #1083 before)~~

:warning:  Redo all tests and notify frontend for travis tests issue.
:warning: Discuss about https://github.com/tracim/tracim/issues/1524#issuecomment-490923517 and https://github.com/tracim/tracim/issues/1524#issuecomment-490930751

### CODE REVIEW

**Comment by inkhey** [comment]
config/doc part is approved by @PhilippeAccorsi but code itself need review.

File: backend/tracim_backend/config.py [hunk 1]
```diff
@@ -788,6 +819,48 @@ class CST(object):
         ASYNC = "ASYNC"
         SYNC = "SYNC"
 
+    def check_mandatory_param(self, param_name: str, value: typing.Any, when_str: str = "") -> None:
+        if not value:
```
Comment (buxx) [inline backend/tracim_backend/config.py]: docstring

File: backend/tracim_backend/config.py [hunk 2]
```diff
@@ -788,6 +819,48 @@ class CST(object):
...
+            raise ConfigurationError(
+                'ERROR: "{}" configuration is mandatory {when_str}.'
+                "Set it before continuing.".format(param_name, when_str="")
+            )
+
+    def check_directory_path_param(
+        self, param_name: str, path: str, writable: bool = False, readable: bool = True
+    ) -> None:
```
Comment (buxx) [inline backend/tracim_backend/config.py]: docstring

File: backend/tracim_backend/command/database.py [hunk 3]
```diff
@@ -119,7 +119,11 @@ def take_action(self, parsed_args: argparse.Namespace) -> None:
                 depot = DepotManager.get()
                 depot_files = depot.list()
                 for file_ in depot_files:
-                    depot.delete(file_)
+                    try:
+                        depot.delete(file_)
+                    # TODO - G.M - 2019-05-09 - better handling of exception here
+                    except Exception:
+                        print("Something goes wrong during deletion of {}".format(file_))
```
[the remark this answers is not in the document]
Comment (inkhey) [reply backend/tracim_backend/command/database.py]: done

File: backend/tracim_backend/config.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/config.py]: done

File: backend/tracim_backend/config.py (hunk 2 above)
Comment (inkhey) [reply backend/tracim_backend/config.py]: done

**Comment by inkhey** [comment]
> @inkhey you delete the database before to drop files. My feeling is, you should do it in reverse order, no? (first drop files - which may fail, then if everything is ok drop the database. Otherwise, You have no reverence to files anymore for deleting content).

Do the other way is also not really nice, it means you have content without associated file which mean content "broken" in tracim.

**Comment by lebouquetin** [comment]
I agree that it's a completely different topic (so you can create a specific issue)
The way it is currently done, we keep files but we have no way to know what they are / what they contain (while having a dtabase without files content is also broken but understandable)

The right way to do things would probably to generate a file listing file which have not been deleted. (to discuss when fixing the specific issue you'll add)

**Comment by inkhey** [comment]
> I agree that it's a completely different topic (so you can create a specific issue)

https://github.com/tracim/tracim/issues/1863

**Pull request merged**